libyul/AsmParser.h

```diff
@@ -119,6 +119,12 @@ class Parser: public langutil::ParserBase
 	/// Creates a DebugData object with the correct source location set.
 	std::shared_ptr<DebugData const> createDebugData() const;
 
+	[[nodiscard]]
+	std::shared_ptr<DebugData const> updateLocationEndFrom(
```

Comment:
Instead of a nodiscard, we could also change the first parameter to  be a non-const reference.

Reply:
ok, I guess it makes sense given the name of the function.